frontend/model/contracts/identity.js

```diff
@@ -20,13 +21,19 @@ sbp('chelonia/defineContract', {
   },
   actions: {
     'gi.contracts/identity': {
-      validate: objectMaybeOf({
-        attributes: objectMaybeOf({
-          username: string,
-          email: string,
-          picture: string
-        })
-      }),
+      validate: (data, { state, meta }) => {
+        objectMaybeOf({
+          attributes: objectMaybeOf({
+            username: string,
+            email: string,
+            picture: string
+          })
+        })(data)
+
+        if (!noUppercase(data.attributes.username)) {
+          throw new TypeError(L('The username cannot contain uppercase letters.'))
```

Comment:
Oh, this actually doesn't need to be wrapped in `L`, since most likely this won't be a user-facing string. If you do want to wrap it in `L` though, I think it makes sense to keep it the same sentence as the other one you have `SignupForm`, so that translators only need to translate one version.

Reply:
Got U. Have removed the L function from it.